File: HeadFirstDesignPatternsJava/HeadFirst/src/StrategyPattern/Duck/MallardDuck.java
package StrategyPattern.Duck;

import StrategyPattern.Fly.FlyWithWings;
import StrategyPattern.Quack.Quack;

public class MallardDuck extends Duck {

    public MallardDuck() {
        super();
        this.flyBehavior = new FlyWithWings();
        this.quackBehavior = new Quack();
    }

    @Override
    public void display() {
        System.out.println("I'm a real Mallaard duck");
    }
    
}

File: HeadFirstDesignPatternsOC/StrategyPattern/Duck/MallardDuck.m
//
//  MallardDuck.m
//  StrategyPattern
//
//  Created by 陳峻琦 on 25/5/2022.
//

#import "MallardDuck.h"
#import "FlyWithWings.h"
#import "Quack.h"

@implementation MallardDuck

- (instancetype)init {
    if (self = [super init]) {
        self.flyBehavior = [FlyWithWings new];
        self.quackBehavior = [Quack new];
    }
    return self;
}

- (void)display {
    NSLog(@"I'm a real Mallaard duck");
}

@end
